### DISCUSSION TITLE
can we use bpf2go without go?

### DISCUSSION THREAD

**Discussion by chent1996** [cilium/ebpf#1265] [Q&A]
I need to use bpf2go on a VM but that VM doesn't have go, so I can only install the bpf2go binary from my PC using go install then export the binary to the VM.

So I was wondering why it has to be invoked by go, as it said, bpf2go was inspired by bpftool, then I guess it should be able to run independently.

**Comment by lmb** [comment]
You can `go build` it and then pass the neccessary environment variables yourself.

```
GOPACKAGE=example.com/your/package bpf2go ...
```

P.S.: Hopefully this should make things clearer for the next person: https://github.com/cilium/ebpf/pull/1267

**Reply by chent1996** [reply]
Yes, I tried to use it this way, but there are some other problems, since I'm building the entire project in Bazel, and Bazel has a lot of restrictions. 

To make it short, Bazel requires bpf2go to generate outputs to a specified directory instead of its working directory, and also forbids us to parse environment variables manually as it will overwrite the shell default env which contains PATH to find llvm-strip.

Of course Bazel should be blamed for it's complex rules, but I think why don't we make bpf2go just like bpftool that could be used everywhere.

**Reply by lmb** [reply]
I'm open to making `bpf2go` work for bazel users, as long as developing and testing the library doesn't require a bazel dependency. Are you willing to contribute the work necessary to do that? If yes, it'd be great if you could drop into [our slack channel](https://cilium.slack.com/archives/ebpf-go-dev) to discuss what would have to change.

**Reply by chent1996** [reply]
Yes, I just convinced myself to work on this and try to make some modifications to at least allow me to build this whole thing using Bazel.

**Reply by chent1996** [reply]
Hi, I just created a new PR about this: https://github.com/cilium/ebpf/discussions/1265#discussioncomment-7841576

And it works in my bazel project now :)

**Reply by chent1996** [reply]
I don't know how to join the slack group as it is my first time using slack, and it keeps rejecting me and telling me that my email has to be from one of those two domains..? Is this my problem or does this slack group only allow people from these two organizations?

**Reply by lmb** [reply]
Try https://cilium.herokuapp.com/
